wandb/wandb
# fix(launch): Allow spaces in requirements file, remove duplicate wandb bootstrap file

Fixes WB-11794

Description
-----------
`_wandb_bootstrap.py` was failing to handle the case where a requirement is specified as `xyz @ ...` due to spaces being present. This change removes the spaces, fixing the behavior.

Also removed `sdk/launch/templates/_wandb_bootstrap.py`, since it was a duplicate of `sdk/launch/builder/templates/_wandb_bootstrap.py`

Testing
-------
1. cd to `wandb/sdk/launch/builder/templates/`
2. Create a `requirements.frozen.txt` file with the contents:

```
python-package-example
example-pypi-package @ https://files.pythonhosted.org/packages/93/8d/e1e98360dc899e533cb3dd857494f2571b129bdffcee76365009b2bb507c/example_pypi_package-0.1.0.tar.gz
python-example-package
```

3. Run `python _wandb_bootstrap.py`
4. Run `pip list | grep 'example'`, confirm that the three example packages were installed
5. Cleanup: Run `pip uninstall -r requirements.frozen.txt -y`

Also ran `tox -e pylaunch38`
